runtime/drivers/clickhouse/olap.go

```diff
@@ -60,6 +60,25 @@ func (c *connection) Exec(ctx context.Context, stmt *drivers.Statement) error {
 		c.logger.Info("clickhouse query", zap.String("sql", stmt.Query), zap.Any("args", stmt.Args))
 	}
 
+	// We use the meta conn for dry run queries
+	if stmt.DryRun {
+		conn, release, err := c.acquireMetaConn(ctx)
+		if err != nil {
+			return err
+		}
+		defer func() { _ = release() }()
+
+		// TODO: Find way to validate with args
+		name := uuid.NewString()
+		_, err = conn.ExecContext(ctx, fmt.Sprintf("CREATE VIEW %q AS %s", name, stmt.Query))
+		if err != nil {
+			return err
+		}
+
+		_, err = conn.ExecContext(context.Background(), fmt.Sprintf("DROP VIEW %q", name))
+		return err
```

Comment:
Did you consider `EXPLAIN`? Wondering if it would reduce the chance of orphaned resources or flashes in the table listings

Reply:
Good idea. Actually just realised that we should not use `CREATE` for external connections. Having write access is not guaranteed. 
Let me check if `EXPLAIN` has any latency issues.